elastic/kibana
# [Uptime] Disable 'Create Rule' button when user doesn't have uptime write permissions [#118404]

## Summary

This PR closes #118404.

Before this PR, users would be able to open the flyout to create an alert and would end-up seeing an error toast when they tried to save it. This flyout could be opened both from within the the top menu, and the monitor view itself, in the popover for when users have enabled anomaly detection.

This PR adds commits to:

* Disable the create alert button in the menu on the top when the user doesn't have permissions to write to Uptime. It will also display a helpful tooltip.
* Disable the create alert button within the monitor view itself, in the popover for when users have enabled anomaly detection. This disabled button will contain a tooltip too.

Both changes above can be seen in the gifs below (in the same order):

![tooltip-top](https://user-images.githubusercontent.com/6868147/145079644-bf58eb32-e870-4e29-afaa-c2a0fc4ed1ba.gif)

![tooltip-anomaly](https://user-images.githubusercontent.com/6868147/145080204-27dff96d-a672-4035-8243-7963b7e16261.gif)

Furthermore, after this PR users who don't have access to uptime will not see the flyover for creating alerts popup automatically because they can't use it anyway.


## How to this PR:

### Test Case 1: Top Menu - Alert Disabled

1. Create a role which has `All` permissions to use Machine Learning capabilities
2. Create a new user and assign it the role from step (1) and the `viewer` role
3. Login with the user from step (2) and ensure that you cannot use the top menu to create an alert (the button should be disabled) and that a tooltip appears.

### Test Case 2: Top Menu - Alert Enabled

1. Create a monitor and make sure it has run
2. As a user with write access to Uptime, try use the button on the top right corner to create an alert. The button should be enabled and the alert creation should succeed.
 
### Test Case 3: Monitor View - Alert Disabled

1. Create a monitor and make sure it has run
2. Create a role which has `All` permissions to use Machine Learning capabilities
3. Create a new user and assign it the role from step (2) and the `viewer` role
4. Login with the user from step (2) and access the details page for the monitor from step (1)
5. Try to create a new anomaly detection rule and ensure the alert flyout does _not_ pop automatically.
6. Ensure that when clicking the anomaly detection button you cannot create an alert (the button should be disabled) and that a tooltip appears.

### Test Case 4: Monitor View - Alert Enabled

1. Create a monitor and make sure it has run
2. As a user with write access to Uptime, try to create a new anomaly detection rule and ensure the alert flyout does pop automatically.
3. Ensure that the button to create an alert in the "Anomaly detection" is enabled and that you can create an alert through it


### Checklist

Delete any items that are not applicable to this PR.



### For maintainers


## Release note

Disables the button to create alerts in Uptime when users do not have permissions to do so.
